aws-sdk-resources/lib/aws-sdk-resources/services/s3/bucket.rb

```diff
@@ -24,8 +24,21 @@ def clear!
       #
       # @return [void]
       def delete!
-        clear!
-        delete
+
+        attempts = 0
+
+        begin
+          clear!
+          delete
+        rescue Errors::BucketNotEmpty => e
+          Kernel.sleep 1.3 ** attempts
+
+          fail e if attempts > 20
```

Comment:
This is a fairly lengthy retry period, I'd like to see this start with a tighter default (like 3), with an option that can be passed in to set your own retry period.